File: bsp4kt/src/com/jetbrains/bsp/bsp4kt/ScalaBuildServer.kt
package com.jetbrains.bsp.bsp4kt

import com.jetbrains.jsonrpc4kt.services.JsonNotification
import com.jetbrains.jsonrpc4kt.services.JsonRequest
import kotlinx.serialization.SerialName
import java.util.concurrent.CompletableFuture

interface ScalaBuildServer {
@JsonRequest("buildTarget/scalacOptions")
fun buildTargetScalacOptions(params: ScalacOptionsParams): CompletableFuture<ScalacOptionsResult>

@Deprecated("Use buildTarget/jvmTestEnvironment instead")
@JsonRequest("buildTarget/scalaTestClasses")
fun buildTargetScalaTestClasses(params: ScalaTestClassesParams): CompletableFuture<ScalaTestClassesResult>

@Deprecated("Use buildTarget/jvmRunEnvironment instead")
@JsonRequest("buildTarget/scalaMainClasses")
fun buildTargetScalaMainClasses(params: ScalaMainClassesParams): CompletableFuture<ScalaMainClassesResult>

}

File: bsp4rs/src/com/jetbrains/bsp/bsp4rs/scala/scala_build_server.rs
use crate::*;

#[derive(Debug)]
pub enum BuildTargetScalacOptions {}

impl Request for BuildTargetScalacOptions {
    type Params = ScalacOptionsParams;
    type Result = ScalacOptionsResult;
    const METHOD: &'static str = "buildTarget/scalacOptions";
}

#[derive(Debug)]
pub enum BuildTargetScalaTestClasses {}

impl Request for BuildTargetScalaTestClasses {
    type Params = ScalaTestClassesParams;
    type Result = ScalaTestClassesResult;
    const METHOD: &'static str = "buildTarget/scalaTestClasses";
}

#[derive(Debug)]
pub enum BuildTargetScalaMainClasses {}

impl Request for BuildTargetScalaMainClasses {
    type Params = ScalaMainClassesParams;
    type Result = ScalaMainClassesResult;
    const METHOD: &'static str = "buildTarget/scalaMainClasses";
}
